coil-base/src/main/java/coil/fetch/HttpUriFetcher.kt

```diff
@@ -221,15 +228,18 @@ internal class HttpUriFetcher(
 
     private fun DiskCache.Snapshot.toCacheResponse(): CacheResponse? {
         try {
-            return metadata.source().buffer().use(::CacheResponse)
+            return fileSystem.read(metadata) {
+                CacheResponse(this)
+            }
         } catch (_: IOException) {
             // If we can't parse the metadata, ignore this entry.
             return null
         }
     }
 
     private fun DiskCache.Snapshot.toImageSource(): ImageSource {
-        return ImageSource(file = data, diskCacheKey = diskCacheKey, closeable = this)
+        // TODO discuss supporting Path in ImageSource as well
+        return ImageSource(file = data.toFile(), diskCacheKey = diskCacheKey, closeable = this)
```

Comment:
I hadn't thought about it, but it probably makes sense to have `ImageSource` use Okio's filesystem as well. Though, internally in one of the implementations we use `File.createTempFile` which only works on the system filesystem so we'll have to figure out what to do that. Let's leave it as is in this PR for now. What do you think?

Fortunately, `ImageSource` is a sealed class so we can change the implementations to use Okio's filesystem and expose `Path`s without breaking binary compatibility later if needed.